### PR TITLE
Rebase all images on Red Hat Universal Base Image

### PR DESCRIPTION
**Pull request by mmanciop**
* Delete Ubuntu-based dynamic Docker image
* Make the previously-named `rhel` image (which was already UBI-based) into the `dynamic` image
* Rebase static images for x64 and s390x on UBI
* Unify Dockerfiles for different architectures
* Adopt Docker buildkit to use build secrets (specifically: the download key)

### CODE REVIEW

**Comment by mdonkers** [comment]
FYI, Conflict probably comes from merging this PR https://github.com/instana/instana-agent-docker/pull/52,
introducing `KUBERNETES` Agent Mode.

File: dynamic/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,112 @@
...
+
+RUN microdnf install --nodocs iptables iproute util-linux python3 jq findutils \
+    && rm -rf /tmp/* \
+    && microdnf clean all
+
+# s390x does not carry the Docker cli
+RUN ( \
+        [[ "${TARGETPLATFORM}" != 'linux/s390x' ]] \
+        && echo -e '[docker-ce-stable]\nname=Docker CE Stable - $basearch\nbaseurl=https://download.docker.com/linux/centos/8/$basearch/stable\nenabled=1\ngpgcheck=1\ngpgkey=https://download.docker.com/linux/centos/gpg' > /etc/yum.repos.d/docker.repo \
+        && microdnf install --nodocs docker-ce-cli tar gzip wget \
```
Comment (mdonkers) [inline dynamic/Dockerfile:60]: Why is for `static` also `containerd.io` getting installed but not for `dynamic`?
See `static/Dockerfile` line 60
`        && microdnf install --nodocs docker-ce-cli containerd.io \`

**Review by mdonkers — COMMENTED** [review]
`KUBERNETES` needs to be put back as `INSTANA_AGENT_MODE` possible value. (might get resolved by merging `master`..

File: dynamic/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,112 @@
...
-    apt-get install -y --no-install-recommends gomplate && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
+# syntax=docker/dockerfile:experimental
+
+FROM --platform=linux/amd64 registry.access.redhat.com/ubi8/ubi-minimal:latest AS build-gomplate
+
+ENV LANG=C.UTF-8 \
+    GOMPLATE_VERSION="3.8.0" \
+    GOMPLATE_SHA256="d32817821a7b083b0aeadf29c2aea17e3384b6b619f3a6285150db1914b94c89  v3.8.0.tar.gz"
```
Comment (mdonkers) [inline dynamic/Dockerfile:7]: If we want to keep up with the latest, `3.9.0` was released a few days ago :-)

File: static/Dockerfile [hunk 3]
```diff
@@ -1,29 +1,112 @@
...
+        && microdnf install --nodocs docker-ce-cli containerd.io \
+        && rm -rf /tmp/* /etc/yum.repos.d/docker.repo \
+        && microdnf clean all \
+    ) || true
+
+# containerd cli is only available for amd64
+ENV CONTAINERD_VERSION=1.4.3 \
+    CONTAINERD_SHA256="34a161e3f459fd337b03141a339eeb1a56c5c811922fe72012d2dac9fa5542f1  containerd-1.4.3-linux-amd64.tar.gz"
+RUN ( \
+        [[ "${TARGETPLATFORM}" == 'linux/amd64' ]] \
```
Comment (mdonkers) [inline static/Dockerfile:69]: **nitpick**, single `=` for consistency (see dynamic)
```suggestion
        [[ "${TARGETPLATFORM}" = 'linux/amd64' ]] \
```

**Comment by mdonkers** [comment]
Did a full review. Nothing special / major, mostly some questions and nitpicks.
Really nice stuff :+1:

File: dynamic/Dockerfile (hunk 1 above)
Comment (cedricziel) [reply dynamic/Dockerfile:60]: Good catch!

**Comment by cedricziel** [comment]
> `KUBERNETES` needs to be put back as `INSTANA_AGENT_MODE` possible value. (might get resolved by merging `master`..

merged master.

**Review by mdonkers — APPROVED** [review]
:tada: LGTM

File: dynamic/README.md [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
-*Note*
+## Building
 
-FTP_PROXY is being abused to pass in the agent key for the package download during docker build, we are doing this until docker build time secrets issue is resolved: [issue GH33343](https://github.com/moby/moby/issues/33343)
+Needs Docker 18.09 or higher:
 
-Docker Hub
-==========
+```sh
+export TARGETPLATFORM=linux/s390x && docker buildx build --no-cache --build-arg DOWNLOAD_KEY=nqtbV5cEQ5ev0MFzOIwskg "--platform=${TARGETPLATFORM}" --build-arg "TARGETPLATFORM=${TARGETPLATFORM}" . -t instana/agent
```
Comment (mdonkers) [inline dynamic/README.md:16]: !!! Should this `DOWNLOAD_KEY` not be some dummy or placeholder... ???

Comment (mdonkers) [reply dynamic/README.md:16]: Also the rest of this command does not appear to be easily copy-pastable. It has some issues.

File: README.md [hunk 5]
```diff
@@ -1,17 +1,8 @@
...
+This repo contains the Dockerfiles relating to the various Instana agent containers.
 
- * dynamic - is the image used in the Helm chart and is suggested as the
-             preferred image for a standard install.
- * static - the static image that contains all run-time dependencies ideal for
-            network access restricted environments such as airgapped servers.
- * rhel - is the RHEL Atomic based image which is suggested for environments
-          where a RHEL base image is required.
+* [`dynamic`](./dynamic/) is the image used by default in the Helm chart and is suggested as the preferred image for a standard install; the `dynamic` image container a dynamic Instana agent, which is capable of updating itself automatically as new versions of its components are published.
+* [`static`](./dynamic/) is an image that contains all run-time dependencies ideal for network access restricted environments such as airgapped servers.
```
Comment (mdonkers) [inline README.md:6]:
```suggestion
* [`static`](./static/) is an image that contains all run-time dependencies ideal for network access restricted environments such as airgapped servers.
```

**Review by mdonkers — CHANGES_REQUESTED** [review]
The enabling of the memory calculator seems to break the images. We should not merge until fully analyzed.

**Review by mdonkers — APPROVED** [review]
Memory calculator disabled.
Reviewed and tested once more and good to go now.

**Pull request merged**